internal/core/src/exec/expression/GISFunctionFilterExpr.cpp

```diff
@@ -143,10 +143,194 @@ PhyGISFunctionFilterExpr::EvalForDataSegment() {
+                    hit_offsets.reserve(coarse.count());
+                    for (size_t i = 0; i < coarse.size(); ++i) {
+                        if (coarse[i]) {
+                            hit_offsets.emplace_back(static_cast<int64_t>(i));
+                        }
+                    }
+                    return hit_offsets;
+                };
+
+                // Lambda: Process sealed segment data using bulk_subscript
+                auto process_sealed_data =
+                    [&](const std::vector<int64_t>& hit_offsets) {
+                        if (hit_offsets.empty())
+                            return;
+
+                        auto data_array = segment_->bulk_subscript(
+                            field_id_, hit_offsets.data(), hit_offsets.size());
+
+                        auto geometry_array = static_cast<
+                            const milvus::proto::schema::GeometryArray*>(
+                            &data_array->scalars().geometry_data());
+                        const auto& valid_data = data_array->valid_data();
+
+                        for (size_t i = 0; i < hit_offsets.size(); ++i) {
+                            const auto pos = hit_offsets[i];
+
+                            // Skip invalid data
+                            if (!valid_data.empty() && !valid_data[i]) {
+                                continue;
+                            }
+
+                            const auto& wkb_data = geometry_array->data(i);
+                            Geometry left(wkb_data.data(), wkb_data.size());
+
+                            if (evaluate_geometry(left)) {
+                                refined.set(pos);
+                            }
+                        }
+                    };
+
+                auto hit_offsets = collect_hits();
+                process_sealed_data(hit_offsets);
+
+                // Cache refined result for reuse by subsequent batches
+                cached_index_chunk_id_ = i;
+                cached_index_chunk_res_ =
+                    std::make_shared<TargetBitmap>(std::move(refined));
+            }
+
+            auto size = ProcessIndexOneChunk(batch_result,
+                                             batch_valid,
+                                             i,
+                                             *cached_index_chunk_res_,
+                                             coarse_valid_global_,
+                                             processed_rows);
+            processed_rows += size;
+            if (processed_rows + size >= real_batch_size) {
+                current_index_chunk_ = i;
+                current_index_chunk_pos_ = i == current_index_chunk_
+                                               ? current_index_chunk_pos_ + size
+                                               : size;
+                break;
+            }
+        }
+    } else {
+        const Index& idx_ref = segment_->chunk_scalar_index<std::string>(
```

Comment:
why we not cache for growing?

for each query, why can't growing result being cached